### PR TITLE
[Editor] Add some telemetry for the highlight feature (bug 1866437)

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]

File: src/display/editor/highlight.js [hunk 1]
```diff
@@ -89,6 +92,34 @@ class HighlightEditor extends AnnotationEditor {
...
+  get telemetryFinalData() {
+    return {
+      type: this.#telemetryType,
+      color: this._uiManager.highlightColorNames.get(this.color),
+    };
+  }
+
+  static computeTelemetryFinalData(data) {
+    // We want to know how many colors have been used.
+    return { numberOfColors: data.get("color").size };
```
Comment (Snuffleupagus) [inline src/display/editor/highlight.js:116]: This "loses" the `type`-property that was added in the `telemetryFinalData`-method, but given that this is a static method I suppose that's basically expected and not an issue in this case (e.g. when using this data on the platform side)?

**Review by Snuffleupagus — APPROVED** [review]
r=me, with one more question; thank you!

File: src/display/editor/highlight.js (hunk 1 above)
Comment (calixteman) [reply src/display/editor/highlight.js:116]: Not really:
We still have it here:
https://github.com/mozilla/pdf.js/pull/17737/files#diff-e213953b4778cea74b8773032792a442c5471535524885040282dbda3850a80fR246

The final structure we pass when printing/saving looks like:
```
{
  type: "print",
  data: {
             "highlight": { numberOfColors: 3 },
             "free_highlight": { numberOfColors: 1},
...
```

and the code in the other side looks like:
```
if (["print", "save"].includes(data.type)) {
      Glean.pdfjs.editing[data.type].add(1);
      if (!data.stats) {
        return;
      }
      for (const [key, value] of Object.entries(data.stats)) {
        switch (key) {
          case "highlight":
          case "free_highlight":
            Glean.pdfjsEditingHighlight[data.type].add(1);
            Glean.pdfjsEditingHighlight.numberOfColors.set(
              value.numberOfColors
            );
            break;
        }
      }
      return;
    }
```

**Pull request merged**